app/src/main/java/org/listenbrainz/android/presentation/features/yim/screens/components/YimShareButton.kt

```diff
@@ -55,7 +56,7 @@ fun YimShareButton(
 private fun ShowDialog(
     onDismiss: () -> Unit,
     viewModel: YimViewModel,
-    typeOfImage: Array<String>,
+    typeOfImage: Array<YimShareables>,
```

Comment:
Why is this an array?

Reply:
I remember the dialog being separate for the artists and tracks. But I believe we should have another approach for this rather than send an array.

Reply:
Used array because its fast and dialogs are same everywhere wdym?

Reply:
I had to pass an array because pf the segmentation of all the composables. This array also determines whether to show the share button or not. Empty array means no share button.

Reply:
Just had a look at the ShowDialog function and thought we could make it more generic. But works fine for me now.

Reply:
The dialog has special colorScheme and also the desgin is bad. :P
Will create a component for dialogs soon which we can use Appwide.

Reply:
Aligned

Reply:
?